e2e_playwright/st_popover_test.py

```diff
@@ -63,6 +64,56 @@ def test_popover_button_rendering(
     )
 
 
+def test_popover_width_content(app: Page, assert_snapshot: ImageCompareFunction):
+    """Test popover button with width=content."""
+    content_width_container = get_element_by_key(app, "test_width=content")
+    content_width_popover = open_popover(app, "popover 10 (width=content)")
+    expect(content_width_popover.get_by_test_id("stMarkdown")).to_have_text(
+        "Content width"
+    )
+    assert_snapshot(
+        content_width_container,
+        name="st_popover-width_content",
+    )
+
+
+def test_popover_width_stretch(app: Page, assert_snapshot: ImageCompareFunction):
+    """Test popover button with width=stretch."""
+    stretch_width_container = get_element_by_key(app, "test_width=stretch")
+    stretch_width_popover = open_popover(app, "popover 11 (width=stretch)")
+    expect(stretch_width_popover.get_by_test_id("stMarkdown")).to_have_text(
+        "Stretch width"
+    )
+    assert_snapshot(
+        stretch_width_container,
+        name="st_popover-width_stretch",
+    )
+
+
+def test_popover_width_fixed(app: Page, assert_snapshot: ImageCompareFunction):
+    """Test popover button with width=500px."""
+    fixed_width_container = get_element_by_key(app, "test_width=500px")
+    fixed_width_popover = open_popover(app, "popover 12 (width=500px)")
+    expect(fixed_width_popover.get_by_test_id("stMarkdown")).to_have_text("500px width")
+    assert_snapshot(
+        fixed_width_container,
+        name="st_popover-width_500px",
+    )
+
+
+def test_popover_columns(app: Page, assert_snapshot: ImageCompareFunction):
+    """Test popover buttons in columns."""
+    columns_container = get_element_by_key(app, "test_columns")
+    columns_popover_1 = open_popover(app, "popover 16 (in column 1)")
+    expect(columns_popover_1.get_by_test_id("stMarkdown")).to_have_text(
+        "Popover in column 1"
+    )
```

Comment:
nit: `expect_markdown(columns_popover_1,  "Popover in column 1")` from app_utils might also work here (same for some other cases here).